### ISSUE TITLE
Request to adyen/process/result for payment method iDEAL occasionally fails

### ISSUE THREAD

**Issue by fahu** [Adyen/adyen-magento2#1115]
**Describe the bug**
Payment method: iDEAL

Occasionally (around 1 out 8 orders) the request to adyen/process/result fails with "Order does not exists with increment_id". However the order gets successfully created (but the customer sees an error message as the endpoint returns a 500 error).

**To Reproduce**
Checkout with iDEAL from Netherlands. More details are unfortunately not available.

**Expected behavior**
No exception is thrown -> The endpoint should redirect the user to the order successful page.

**Magento version**
2.3.5

**Plugin version**
6.6.8

**Screenshots**
![image](https://user-images.githubusercontent.com/2507037/126703790-7f144b0a-1b40-4065-a02f-d0b34003b2da.png)

**Additional context**
The issue occurs on different devices (iOS, Android, Desktop Windows with Chrome, ..). 

Might be related to this issue: https://github.com/Adyen/adyen-magento2/issues/717

**Comment by acampos1916** [comment]
Hey @fahu, did you mean to close the issue, was #717 helpful to figure out what was wrong?

I see that the merchant reference that is being checked in that location in 6.6.8 comes from the /payments/details call that is made here https://github.com/Adyen/adyen-magento2/blob/6.6.8/Controller/Process/Result.php#L433 so maybe that request is failing in your case. This would be evident in var/log/adyen/info.log.

Let us know if this info helps.

**Comment by fahu** [comment]
hey @acampos1916, thanks for reaching back and sorry for not providing the reason for closing the case.

We actually figured out that this one is most probably caused by an adaption on our side. Wer are currently validating this  and therefore I closed it to not cause any work on your side in case it's really due to our modifications.

Short description of the issue: We used `_getOrder()` in `result.php` to log some additional context and didn't consider the side effect of the method: The value is cached and if its retrieved too early for iDEAL payment (without executing `validatePayloadAndReturnResponse` before `_getOrder()`) an empty order is cached by `_getOrder()`.

Have a nice weekend!
Fabian
